lib/pure/net.nim

```diff
@@ -85,7 +85,11 @@ when defineSsl:
       CVerifyNone, CVerifyPeer
 
     SslProtVersion* = enum
-      protSSLv2, protSSLv3, protTLSv1, protSSLv23
+      protSSLv2 {.deprecated.}
+      protSSLv3 {.deprecated.}
+      protTLSv1 {.deprecated.}
+      protSSLv23 {.deprecated.}
+      protTLS
```

Comment:
I think we should deprecate this enum altogether. I left it thinking we might add protTLSv11, protTLSv12, ... later, but OpenSSL v1.1 has a method to set min and max version which would be a lot more flexible, although making a shim for v1.0 is a pain.